File: inventory-supplier-service/src/main/java/com/cg/inventorysupplierservice/service/SupplierService.java
/**
 * @author Gagandeep Singh
 * @email singh.gagandeep3911@gmail.com
 * @create date 2020-11-02 01:40:33
 * @modify date 2020-11-02 01:40:33
 * @desc Supplier Bussiness Logi Interface
 */
package com.cg.inventorysupplierservice.service;

import java.util.List;

import com.cg.inventorysupplierservice.dto.SupplierDto;
import com.cg.inventorysupplierservice.entity.Supplier;

public interface SupplierService {
  Supplier addSupplier(SupplierDto supplier);

  List<Supplier> getSuppliers();

  Supplier getSupplierById(Long id);
}

File: inventory-frontend/src/app/services/supplier.service.ts
/**
 * @author Gagandeep Singh
 * @email singh.gagandeep3911@gmail.com
 * @create date 2020-11-08 18:35:38
 * @modify date 2020-11-08 18:35:38
 * @desc Manage Supplier related operation
 */
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root',
})
export class SupplierService {
  serviceUrl =
    environment.protocol +
    environment.applicationUrl +
    '/' +
    environment.supplierService;

  constructor(private http: HttpClient) {}

  fetchAllSuppliers() {
    return this.http.get(this.serviceUrl + '/supplier');
  }

  addSupplier(formData) {
    return this.http.post(this.serviceUrl + '/supplier', formData);
  }

  updateSupplier(formData) {
    return this.http.put(this.serviceUrl + '/supplier', formData);
  }

  fetchById(id) {
    return this.http.get(this.serviceUrl + '/supplier/' + id);
  }
}
